py/runai_model_streamer/runai_model_streamer/file_streamer/requests_iterator.py

```
@@ -49,67 +73,137 @@ def append(self, file_chunks: FileChunks) -> None:
+                # don't OOM during allocation. If the model is larger than free VRAM,
+                # the buffer is reused in chunks just like the limited-memory path.
+                memory_limit = min(total_size, free_cuda_memory)
+            else:
+                memory_limit = total_size
         elif memory_mode == MemoryCapMode.largest_chunk:
-            memory_limit = max(max(file_chunks.chunks) for file_chunks in files_chunks)
+            memory_limit = max(max(file_chunks.effective_strides) for file_chunks in files_chunks)
         elif memory_mode == MemoryCapMode.limited:
             if user_memory_limit is None:
                 raise RunaiStreamerMemoryLimitException(
                     f"MemoryCapMode is Limited, but no limit supplied"
```

Ruff (0.15.6)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)